subsys/bluetooth/shell/iso.c

```diff
@@ -164,33 +215,48 @@ static int cmd_cig_create(const struct shell *sh, size_t argc, char *argv[])
 
 	err = 0;
 	if (argc > 2) {
-		unsigned long interval;
+		long interval;
 
 		interval = shell_strtoul(argv[2], 0, &err);
-		if (err != 0) {
-			shell_error(sh, "Could not parse interval: %d", err);
-
-			return -ENOEXEC;
+		interval = parse_interval(sh, argv[2]);
+		if (interval < 0) {
+			return interval;
 		}
 
-		if (!IN_RANGE(interval,
-			      BT_ISO_SDU_INTERVAL_MIN,
-			      BT_ISO_SDU_INTERVAL_MAX)) {
-			shell_error(sh, "Invalid interval %lu", interval);
+		param.c_to_p_interval = interval;
+	} else {
+		param.c_to_p_interval = 10000;
+	}
+
+	if (argc > 3) {
+		long interval;
 
-			return -ENOEXEC;
+		interval = parse_interval(sh, argv[3]);
+		if (interval < 0) {
+			return interval;
 		}
 
-		param.interval = interval;
+		param.p_to_c_interval = interval;
 	} else {
-		param.interval = 10000;
+		param.p_to_c_interval = param.c_to_p_interval;
 	}
-	cis_sdu_interval_us = param.interval;
 
-	if (argc > 3) {
+	/* cis_sdu_interval_us is used to increase the sequence number.
+	 * If we can not get the role information we also should get a
+	 * CIS create error, so we do not need to set this variable.
+	 */
+	err = bt_conn_get_info(default_conn, &info);
+	if (err != 0) {
+		shell_error(ctx_shell, "Failed to get connection information: %d", err);
+	} else {
+		cis_sdu_interval_us = (info.role == BT_CONN_ROLE_CENTRAL) ?
+	  		              param.c_to_p_interval : param.p_to_c_interval;
```

Comment:
Actually in `cmd_cig_create` we will always be the central (it's not allowed to create a CIG as the peripheral)

Maybe the role check should be put further up, and return early if not the central? 
Then this can just be `cis_sdu_interval_us = param.c_to_p_interval`.

The requirement to be a central in `HCI_LE_Set_CIG_Parameters` is arguably kind of weird, as the CIG can be created before any ACL connections...